test/lit/passes/local-cse-cont.wast

```diff
@@ -36,5 +43,36 @@
    )
   )
  )
+
+ ;; CHECK:      (func $cont.bind (type $4) (param $cont-i32 (ref $cont-i32))
+ ;; CHECK-NEXT:  (drop
+ ;; CHECK-NEXT:   (cont.bind $cont-i32 $cont
+ ;; CHECK-NEXT:    (i32.const 42)
+ ;; CHECK-NEXT:    (local.get $cont-i32)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (drop
+ ;; CHECK-NEXT:   (cont.bind $cont-i32 $cont
+ ;; CHECK-NEXT:    (i32.const 42)
+ ;; CHECK-NEXT:    (local.get $cont-i32)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT: )
+ (func $cont.bind (param $cont-i32 (ref $cont-i32))
+  ;; We cannot optimize here: Each of these has a side effect of modifying the
+  ;; continuation they were given (it will trap if resumed).
```

Comment:
I suppose the second cont.bind here should trap without ever having to resume, right?

Reply:
Yes, confirmed this should trap: https://github.com/WebAssembly/stack-switching/blob/main/proposals/stack-switching/Explainer.md#consuming-continuations

Let's update the comment to say that optimizing would inappropriately remove the trap on the second cont.bind.

(It would also be great to add tests for this)

Reply:
Yes, but "use" isn't defined... And, emphasizing the first part of the quote,

> *In order to ensure that continuations are one-shot,* resume, resume_throw, switch, and cont.bind destructively modify the suspended continuation such that any subsequent use of the same suspended continuation will result in a trap.

One-shot continuations are generally defined as ones that can only be **executed** once. But `cont.bind` does not execute it.

Anyhow, maybe this is spec lawyering, sorry, but it did not strike me as unambiguous. But if it's just me, doesn't matter I suppose...

Reply:
Yeah, the given rationale you highlighted is perhaps misleading, but the actual description of the behavior is clear, I think. It makes sense for `cont.bind` to make its input continuation unusable 1) for consistency with the other three instructions and 2) because `cont.bind` is equivalent to resuming the continuation with just some of its expected values and immediately suspending again, and that equivalence only holds if the input continuation is rendered unusable.